kitsune/wiki/forms.py

```diff
@@ -276,6 +276,24 @@ def save(self, creator, document, based_on_id=None, base_rev=None,
         return new_rev
 
 
+class DraftRevisionForm(forms.ModelForm):
+    class Meta(object):
+        model = DraftRevision
+        fields = ('keywords', 'summary', 'content', 'slug', 'title', 'based_on')
+
+    def save(self, request):
+        """save the draft revision and return the draft revision"""
+        creator = request.user
+        doc_data = self.cleaned_data
+        parent_doc = doc_data['based_on'].document
+        locale = request.LANGUAGE_CODE
+        draft, created = DraftRevision.objects.update_or_create(creator=creator,
+                                                                parent_doc=parent_doc,
+                                                                locale=locale,
+                                                                defaults=doc_data)
```

Comment:
This matches our style guide, but personally I think it is more readable like this:

```python
draft, created = DraftRevision.objects.update_or_create(
    creator=creator, parent_doc=parent_doc, locale=locale, defaults=doc_data)
```